sympy/functions/elementary/hyperbolic.py

```diff
@@ -336,12 +346,25 @@ def _eval_as_leading_term(self, x):
             return self.func(arg)
 
     def _eval_is_real(self):
-        return self.args[0].is_real
+        from sympy import pi
+        arg = self.args[0]
+
+        # `cosh(x)` is real for real and purely imaginary `x`
```

Comment:
Was there a case when SymPy couldn't determine if values of the one of functions you have modified was real and your pull request fixed it? That's the case that should be checked in these tests to be written - maybe one for argument in which function is real and another one for argument in which it isn't real for each of modified functions?

It's just a suggestion - let the mentor decide if it's necessary. It would be good to have such test, if you have time to implement it, but again - I don't know how demanding should we be here.